### PR TITLE
feat: replace debian_package_manager with apt

### PR DESCRIPTION
**Pull request by thesayyn**
This the effort for replacing debian package manager with https://github.com/GoogleContainerTools/rules_distroless/pull/16

And a follow up to #1541

### CODE REVIEW

File: .github/workflows/update-deb-package-snapshots.yml [hunk 1]
```diff
@@ -15,27 +15,27 @@ jobs:
...
+      - name: Update snapshots and lock using "knife"
+        run: ./knife update-snapshots && ./knife lock
 
-    - name: Run update script
-      run: ./updateWorkspaceSnapshots.sh
+      - name: Run update sboms script
+        run: |
+          for i in $(seq 5); do
+            bazel --bazelrc=$GITHUB_WORKSPACE/.github/workflows/ci.bazelrc fetch //private/pkg/test/oci_image:test_sboms && break || sleep 20;
+          done
```
Comment (thesayyn) [inline .github/workflows/update-deb-package-snapshots.yml:30]: Auto format :(

File: README.md [hunk 2]
```diff
@@ -217,16 +223,15 @@ Distroless tracks the upstream Debian releases, using [Github actions to automat
 
 ### Debug Images
 
-Distroless images are minimal and lack shell access.  The ```:debug``` image set for each language provides a busybox shell to enter.
+Distroless images are minimal and lack shell access. The `:debug` image set for each language provides a busybox shell to enter.
```
Comment (thesayyn) [inline README.md:226]: @loosebazooka we should probably run pre-commit on the whole repo and rebase here. Can you do that?

File: knife [hunk 3]
```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bash
```
Comment (loosebazooka) [inline knife:1]: I think I would prefer these be separate scripts that had more descriptive names rather than knife?

**Review by loosebazooka — APPROVED** [review]
LGTM, we can make that knife change in a followup?

File: knife (hunk 3 above)
Comment (thesayyn) [reply knife:1]: naming is tbd, i can split them as well.

**Comment by thesayyn** [comment]
> LGTM, we can make that knife change in a followup?

Yes, i'll remove it from here.

**Comment by thesayyn** [comment]
Oh, actually we can't as there's no replacement for what debian_package_manger did to upgrade snapshots etc. now those logic lives in knife.

**Pull request merged**